pkg/controllers/management/auth/project_cluster/common.go

```diff
@@ -141,6 +147,56 @@ func createMembershipRoles(obj runtime.Object, crClient crbacv1.ClusterRoleContr
 	return nil
 }
 
+// checkPSAMembershipRole creates (if needed) an additional cluster role to grant updatepsa permissions, if needed.
+func checkPSAMembershipRole(obj runtime.Object, crClient crbacv1.ClusterRoleController, prtbLister v32.ProjectRoleTemplateBindingCache, rtLister v32.RoleTemplateCache) error {
+	var resourceName, resourceNamespace, resourceType, context string
+	var annotations map[string]string
+
+	switch v := obj.(type) {
+	case *apisv3.Project:
+		context = projectContext
+		resourceType = apisv3.ProjectResourceName
+		resourceName = v.GetName()
+		resourceNamespace = v.Status.BackingNamespace
+	default:
+		return fmt.Errorf("cannot create membership roles for unsupported type %T", v)
+	}
+
+	ownerRef, err := newOwnerReference(obj)
+	if err != nil {
+		return err
+	}
+
+	psaNeeded, err := needsPSARole(resourceName, resourceNamespace, prtbLister, rtLister)
+	if err != nil {
+		return err
+	}
+
+	var psaRole *rbackv1.ClusterRole
+	if psaNeeded {
+		psaRole = &rbackv1.ClusterRole{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:            name.SafeConcatName(resourceName, context, "psa"),
+				OwnerReferences: []metav1.OwnerReference{ownerRef},
+				Annotations:     annotations,
```

Comment:
this annotations is always empty. Can we remove it? or are we missing some annotations here? I can't find where the aggregation labels are. Are they supposed to be here?